src/Server/NeoServer.Server.Events/Player/PlayerLevelRegressedEventHandler.cs

```diff
@@ -9,27 +9,39 @@
 
 namespace NeoServer.Server.Events.Player;
 
-public class PlayerLevelRegressedEventHandler : PlayerLevelChangeEventHandler
+public class PlayerLevelRegressedEventHandler
 {
-    public PlayerLevelRegressedEventHandler(IGameServer game) : base(game)
-    { }
+    private readonly IGameServer game;
 
-    protected override void SendLevelChangeMessage(SkillType skillType, IConnection connection, int fromLevel, int toLevel)
+    public PlayerLevelRegressedEventHandler(IGameServer game)
+    {
+        this.game = game;
+    }
+
+    public void Execute(IPlayer player, SkillType skillType, int fromLevel, int toLevel)
+    {
+        if (Guard.IsNull(player)) return;
+
+        if (!game.CreatureManager.GetPlayerConnection(player.CreatureId, out var connection)) return;
+
+        SendLevelChangeMessage(skillType, connection, fromLevel, toLevel);
+
+        connection.OutgoingPackets.Enqueue(new PlayerStatusPacket(player));
+
+        connection.Send();
+    }
+
+    protected void SendLevelChangeMessage(SkillType skillType, IConnection connection, int fromLevel, int toLevel)
     {
         if (skillType != SkillType.Level)
             return;
 
         connection.OutgoingPackets.Enqueue(new TextMessagePacket(
             $"You regressed from level {fromLevel} to level {toLevel}.",
             TextMessageOutgoingType.MESSAGE_EVENT_LEVEL_CHANGE));
-    }
-
-    public override void SendLevelChangeMessage(SkillType skillType, IConnection connection, int toLevel)
-    {
-        if (skillType == SkillType.Level) return;
 
         connection.OutgoingPackets.Enqueue(new TextMessagePacket(
-            MessageParser.GetSkillAdvancedMessage(skillType, toLevel),
+            MessageParser.GetSkillRegressedMessage(skillType, toLevel),
```

Comment:
I think the message should be sent only when you have any difference between fromLevel and toLevel.
If they are the same there is no need for showing the message I think, right?

Reply:
Yes, but whoever calls this class is validating this